src/doc/man/cargo-tree.adoc

```diff
@@ -0,0 +1,193 @@
+= cargo-tree(1)
+:idprefix: cargo_tree_
+:doctype: manpage
+:actionverb: Display
+:noall: true
+
+== NAME
+
+cargo-tree - Display a tree visualization of a dependency graph
+
+== SYNOPSIS
+
+`cargo tree [_OPTIONS_]`
+
+== DESCRIPTION
+
+This command will display a tree of dependencies to the terminal. An example of a simple project that depends on the "rand" package:
+
+----
+myproject v0.1.0 (/myproject)
+└── rand v0.7.3
+    ├── getrandom v0.1.14
+    │   ├── cfg-if v0.1.10
+    │   └── libc v0.2.68
+    ├── libc v0.2.68 (*)
+    ├── rand_chacha v0.2.2
+    │   ├── ppv-lite86 v0.2.6
+    │   └── rand_core v0.5.1
+    │       └── getrandom v0.1.14 (*)
+    └── rand_core v0.5.1 (*)
+[build-dependencies]
+└── cc v1.0.50
+----
+
+Packages marked with `(*)` have been "de-duplicated". The dependencies for the
+package have already been shown elswhere in the graph, and so are not
+repeated. Use the `--no-dedupe` option to repeat the duplicates.
+
+== OPTIONS
+
+=== Tree Options
+
+*-i*::
+*--invert*::
+    Invert the tree. Typically this flag is used with the `-p` flag to show
+    the dependencies for a specific package.
+
+*--no-dedupe*::
+    Do not de-duplicate repeated dependencies. Usually, when a package has
+    already displayed its dependencies, further occurrences will not
+    re-display its dependencies, and will include a `(*)` to indicate it has
+    already been shown. This flag will cause those duplicates to be repeated.
+
+*-d*::
+*--duplicates*::
+    Show only dependencies which come in multiple versions (implies
+    `--invert`). When used with the `-p` flag, only shows duplicates within
+    the subtree of the given package.
++
+It can be beneficial for build times and executable sizes to avoid building
+that same package multiple times. This flag can help identify the offending
+packages. You can then investigate if the package that depends on the
+duplicate with the older version can be updated to the newer version so that
+only one instance is built.
+
+*--no-dev-dependencies*::
```

Comment:
Your recommendation of `--dep-kinds` makes sense to me, and I think that'd lend itself pretty nicely here (and avoid the no-* prefix!)

Reply:
I added `--dep-kinds` which supports a variety of combinations.  I'm not 100% happy with it (`--dep-kinds=no-dep` is wordy and awkward).  But I wonder if maybe `--dep-kinds=normal` might be a more common usage, and is maybe a little easier to read?

Reply:
Perhaps `--edges` may make the wording a bit nicer? That's sort of what it actually is (edges in a graph), but `--edges normal` doesn't read quite as nicely as `--dep-kinds normal`